src/core/qgsstringutils.cpp

```diff
@@ -563,6 +563,11 @@ QString QgsStringUtils::insertLinks( const QString &string, bool *foundLinks )
   return converted;
 }
 
+bool QgsStringUtils::isUrl( const QString &string )
+{
+  return QRegularExpression( "^(http|https|ftp|file)://\\S+$" ).match( string ).hasMatch();
```

Comment:
```suggestion
  const thread_local rxUrl QRegularExpression( "^(http|https|ftp|file)://\\S+$" ); 
  return rxUrl.match( string ).hasMatch();
```

Given that this function WILL be called many millions of times in a single QGIS session, it's worth optimising!

Reply:
(needs unit tests too)